src/Dofs/MixedDofHandler.jl

```diff
@@ -240,10 +240,10 @@ function __close!(dh::MixedDofHandler{dim}) where {dim}
     @debug "\n\nCreating dofs\n"
     for fh in dh.fieldhandlers
         # sort the cellset since we want to loop through the cells in a fixed order
-        cellnumbers = sort(collect(fh.cellset))
+        # cellnumbers = issorted(fh.cellset) ? sort(collect(fh.cellset)) : fh.cellset
         nextdof = _close!(
             dh,
-            cellnumbers,
+            fh.cellset,
```

Comment:
If you think that the sorted order is really that essential we can also recommending sorted set or ordered set as data structures.

Reply:
It would be good to have it be more deterministic perhaps. Maybe we should make cellsets in the grid vectors, or keep both a vector version and a set version of the data.

Reply:
Mhh. Maybe we can add the set type as a type parameter to the grid.

Reply:
But then all sets need to have the same type. I am thinking we could either maintain both datastructures, but lazily initialize them as needed. Alternatively, use sorted vectors, and make sets when needed. Probably we can get away with doing it only in a few places such as in `close!` etc.

Reply:
I think it makes sense to have the slow path for Set and the optimized paths for UnitRanges, which I will document as the recommended way to setup the subdomains, if we agree on this. UnitRanges keep the caches hot, are small and all operations are fast. What do you think?

Reply:
We can use `OrderedSet` (after https://github.com/JuliaCollections/OrderedCollections.jl/pull/100) and make sure to `sort!` in `addcellset!` for example.